### PR TITLE
optionally show diff

### PR DESCRIPTION
**Pull request by mpollmeier**
example for a diff:
scala-colordiff

### CODE REVIEW

File: sbt-scalafmt/src/main/scala/com/lucidchart/sbt/scalafmt/ScalafmtCorePlugin.scala [hunk 1]
```diff
@@ -163,15 +170,22 @@ object ScalafmtCorePlugin extends AutoPlugin {
...
+              file =>
+                val original = IO.read(file)
+                val formatted = scalafmtter(file.toString, original)
+                val hasChanges = original != formatted
+                if (hasChanges) {
+                  val msg = if (showDiff) {
+                    val diff = ColorDiff(original.split('\n').toList, formatted.split('\n').toList)
+                    s"$file has changes after scalafmt:\n$diff"
+                  } else s"$file has changes after scalafmt"
+                  logger.warn(msg)
```
Comment (joeycozza) [inline sbt-scalafmt/src/main/scala/com/lucidchart/sbt/scalafmt/ScalafmtCorePlugin.scala]: Is this change going to logger.warn(msg) twice if failForStyleIssues is false?

It looks like if we remove line 184, then we'll have the logging behavior we used to have. Is that right, or am I missing something?

**Review by joeycozza — CHANGES_REQUESTED** [review]

File: sbt-scalafmt/src/main/scala/com/lucidchart/sbt/scalafmt/ScalafmtCorePlugin.scala (hunk 1 above)
Comment (mpollmeier) [reply sbt-scalafmt/src/main/scala/com/lucidchart/sbt/scalafmt/ScalafmtCorePlugin.scala]: thanks for catching that, you're right. I just removed that line.

File: sbt-scalafmt/build.sbt [hunk 2]
```diff
@@ -6,9 +6,11 @@ buildInfoPackage := "com.lucidchart.sbt.scalafmt"
 
 libraryDependencies ++= Seq(
   "com.google.code.findbugs" % "jsr305" % "3.0.2" % Provided, // fixes warnings about javax annotations
-  "com.google.guava" % "guava" % "19.0"
+  "com.google.guava" % "guava" % "19.0",
+  "com.michaelpollmeier" %% "colordiff" % "0.8"
 )
 
+resolvers += Resolver.mavenLocal
```
Comment (joeycozza) [inline sbt-scalafmt/build.sbt]: I was doing 1 more pass of the code before merging, and I am wondering why you added this line? Any particular reason, or could that be removed from the PR? I'm not opposed to keeping it if there is a good reason. (SBT configs and build files are new territory for me)

**Review by joeycozza — CHANGES_REQUESTED** [review]

File: sbt-scalafmt/build.sbt (hunk 2 above)
Comment (mpollmeier) [reply sbt-scalafmt/build.sbt]: that wasn't supposed to get committed, I'll remove it right now.

**Comment by joeycozza** [comment]
Thanks for this. That is a cool feature. I pushed a new version for 1.11 and that should be live pretty soon.

**Pull request merged**